webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -0,0 +1,372 @@
+  ExperimentListSettings,
+  settingsConfigForProject,
+} from '../ExperimentList.settings';
+import css from '../ProjectDetails.module.scss';
+
+import GlideTable from './glide-table/GlideTable';
+import { useGlasbey } from './glide-table/useGlasbey';
+
+interface ExplistProps {
+  project: Project;
+}
+const filterKeys: Array<keyof ExperimentListSettings> = ['label', 'search', 'state', 'user'];
+
+const PAGE_RADIUS = 50;
+const F_ExperimentList: React.FC<ExplistProps> = ({ project }) => {
+  const [pageMidpoint, setPageMidpoint] = useState(0);
+
+  const handleScroll = useCallback(({ y, height }: Rectangle) => {
+    const visibleRegionMidPoint = y + height / 2;
+    setPageMidpoint(Math.round(visibleRegionMidPoint / PAGE_RADIUS) * PAGE_RADIUS);
+  }, []);
+
+  const [experiments, setExperiments] = useState<ExperimentItem[]>([]);
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [canceler] = useState(new AbortController());
+  const pageRef = useRef<HTMLElement>(null);
+  const { width } = useResize(pageRef);
+
+  const id = project?.id;
+
+  const settingsConfig = useMemo(() => settingsConfigForProject(id), [id]);
+
+  const { settings, updateSettings, resetSettings, activeSettings } =
+    useSettings<ExperimentListSettings>(settingsConfig);
+
+  const filterCount = useMemo(() => activeSettings(filterKeys).length, [activeSettings]);
+
+  const statesString = useMemo(() => settings.state?.join('.'), [settings.state]);
+  const pinnedString = useMemo(() => JSON.stringify(settings.pinned ?? {}), [settings.pinned]);
+  const labelsString = useMemo(() => settings.label?.join('.'), [settings.label]);
+  const usersString = useMemo(() => settings.user?.join('.'), [settings.user]);
+
+  const fetchExperiments = useCallback(
+    async (): Promise<void> => {
+      if (!settings) return;
+      try {
+        const states = statesString
+          ?.split('.')
+          .map((state) => encodeExperimentState(state as RunState));
+        // const pinned = JSON.parse(pinnedString);
+        const baseParams: GetExperimentsParams = {
+          archived: settings.archived ? undefined : false,
+          labels: settings.label,
+          name: settings.search,
+          orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
+          projectId: id,
+          sortBy: validateDetApiEnum(V1GetExperimentsRequestSortBy, settings.sortKey),
+          states: validateDetApiEnumList(Experimentv1State, states),
+          users: settings.user,
+        };
+
+        const tableLimit = 2 * PAGE_RADIUS;
+        const tableOffset = Math.max(pageMidpoint - PAGE_RADIUS, 0);
+
+        const expResponse = await getExperiments(
+          {
+            ...baseParams,
+            limit: tableLimit,
+            offset: tableOffset,
+          },
+          { signal: canceler.signal },
+        );
+
+        setExperiments((prevExperiments) => {
+          const newExperiments = [];
+          let i = 0;
+          while (i < tableOffset) {
```

Comment:
im not sure what you mean. something like:

```
const newExperiments = [...prevExperiments.slice(0, tableOffset), ...newExperiments, ...prevExperiments(tableOffset + total)]
```
?

the above doesn't work because you could get into a state where `prevExperiments.slice(0, tableOffset).length < tableOffset` and your indices would get thrown off.

 though the utitity of the current approach hinges on the TODO below